src/main/scala/mesosphere/marathon/io/IO.scala

```diff
@@ -47,21 +45,19 @@ object IO extends StrictLogging {
     * @param mabyeTo Outputstream to copy to.
     * @return
     */
-  def transfer(maybeFrom: Option[InputStream], mabyeTo: Option[OutputStream]): Long = {
-    (maybeFrom, mabyeTo) match {
+  def transfer(maybeFrom: Option[InputStream], maybeTo: Option[OutputStream]): Long = {
+    (maybeFrom, maybeTo) match {
       case (Some(from), Some(to)) =>
-        val buf = createBuffer();
-        var total = 0
-        while (true) {
-          val r = from.read(buf)
-          if (r == -1) {
-            break
+        @tailrec def iter(buf: Array[Byte], total: Long): Long =
+          from.read(buf) match {
+            case -1 => total
+            case r =>
+              to.write(buf, 0, r)
+              to.flush()
```

Comment:
What's the advantage of flushing on every write call, instead of calling it after it is done writing?

Reply:
That was a bug. When we copy an SSE stream the write will never be finished. So we have to flush after each chunk.

Reply:
Got it. Thank you!